app/views/Data/Workspace/index.tsx

```diff
@@ -78,40 +140,73 @@ function WorkspaceItem(props: WorkspaceItemProps) {
                 loading={removeTableFromWorkspaceResult.fetching}
                 leftIcon={<MdOutlineTableChart />}
                 disabled={removeTableFromWorkspaceResult.fetching}
+                onClick={onClickTable}
             >
-                <Text color="dark">
-                    {table.name}
-                </Text>
+                {!isRename
+                    ? (
+                        <Text color="dark">
+                            {table.name}
+                        </Text>
+                    ) : (
+                        <TextInput
+                            autoFocus
+                            className={styles.renameInput}
+                            value={renameTable}
+                            onChange={(e) => setRenameTable(e.currentTarget.value)}
```

Comment:
let's not used anonymous functions. They will always be created on each `render`.